test/cypress/support/command/page-command.js

```diff
@@ -0,0 +1,151 @@
+  }
+  if (friendlyCode) {
+    // eslint-disable-next-line max-len
+    cy.getInputByName(replaceLangCodePlaceholder(TEST_ID_PAGE_FORM.SEO_FRIENDLY_CODE_FIELD, lang))
+      .type(friendlyCode);
+  }
+}
+
+function fillPageForm(page, languages, defaultLang, button) {
+  cy.getInputByName(replaceLangCodePlaceholder(TEST_ID_PAGE_FORM.TITLE_FIELD, defaultLang))
+    .type(page.titles[defaultLang]);
+  if (page.seoData) {
+    fillSeoDataByLangFields(page.seoData.seoDataByLang[defaultLang], defaultLang);
+    cy.addPageDefaultLangMetaTags(page.seoData.seoDataByLang[defaultLang].metaTags);
+  }
+  // remove the default lang from the list of languages
+  const langugesFiltered = languages.filter(item => item !== defaultLang);
+  langugesFiltered.forEach((lang) => {
+    cy.getByTestId(TEST_ID_PAGE_FORM.TABS).contains(lang.toUpperCase()).click();
+    cy.wait(200);
+    cy.getInputByName(replaceLangCodePlaceholder(TEST_ID_PAGE_FORM.TITLE_FIELD, lang))
+      .type(page.titles[lang]);
+    if (page.seoData) {
+      fillSeoDataByLangFields(page.seoData.seoDataByLang[lang], lang);
+      cy.addPageMetaTags(page.seoData.seoDataByLang[lang].metaTags, lang);
+    }
+  });
+  cy.getByTestId(TEST_ID_PAGE_FORM.OWNER_GROUP_TYPEAHEAD).click();
+  cy.wait(500);
+  cy.getTypeaheadOption(page.ownerGroup).click();
+  cy.getInputByName(TEST_ID_PAGE_FORM.CODE_FIELD).clear();
+  cy.getInputByName(TEST_ID_PAGE_FORM.CODE_FIELD).type(page.code);
+  cy.getByTestId(TEST_ID_PAGE_FORM.PAGE_TREE_SELECTOR_FIELD).contains(page.parentPage).click();
+  cy.getByName(TEST_ID_PAGE_FORM.PAGE_TEMPLATE_FIELD).select(page.pageModel);
+  cy.getByTestId(button).click();
+}
+
+/**
+ * Add a new Page
+ * @param page - The page object
+ * @param languages - An array with language codes e.g. ["en", "it"]
+ * @param defaultLang - The code of the default lang (the default value is 'en')
+ * @param button - The button to click ( the default button to click is 'save')
+ */
+Cypress.Commands.add('addPage', (
+  page, languages, defaultLang = 'en',
+  button = TEST_ID_PAGE_FORM.SAVE_BUTTON,
+) => {
+  cy.log(`Add a new page  ${page.code} with default lang ${defaultLang}`);
+  cy.openPageFromMenu(['Pages', 'Management']);
+  cy.getButtonByText('Add').click();
+  cy.validateUrlChanged('/page/add');
+  fillPageForm(page, languages, defaultLang, button);
+});
+
+/**
+ * Delete a Page from the page tree
+ * @param pageCode - the page code to delete
+*/
+Cypress.Commands.add('deletePage', (pageCode) => {
+  cy.log(`Delete page  ${pageCode}`);
+  cy.expandAllPageTree();
+  cy.wait(1000);
+  cy.openTableActionsByTestId(pageCode).then(() => {
+    cy.clickOnTableActionMenu(TEST_ID_LIST_PAGE_TREE.ACTION_MENU, 'Delete');
+  });
+  cy.getModalDialogByTitle('Delete page').should('be.visible');
+  cy.getButtonByText('Delete').click();
+});
+
+/**
+ * Publish a Page from the page tree
+ * @param pageCode - the page code to publish
+ */
+Cypress.Commands.add('publishPageAction', (pageCode) => {
```

Comment:
is the `action` suffix really needed? do we need to know the page was published by clicking on the kebab action or elsewhere?

Reply:
Yes we have another publish page command in page designer (publishPageClick) and I prefer to differ this from the other (action refers to the actions menu)